extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/ReactiveSession.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.hibernate.reactive.panache.common;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.interceptor.InterceptorBinding;
+
+import org.hibernate.reactive.mutiny.Mutiny;
+
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Instructs Panache to invoke the intercepted method within a scope of a reactive {@link Mutiny.Session}.
+ * <p>
+ * An existing reactive {@link Mutiny.Session} is reused. If no session exists then a new session is opened lazily and
+ * eventually closed when the {@link Uni} returned from the annotated method completes.
+ * <p>
+ * A method annotated with this annotation must return {@link Uni}. If declared on a class then all methods that are intercepted
+ * must return {@link Uni}.
+ */
+@Inherited
+@InterceptorBinding
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(value = RetentionPolicy.RUNTIME)
+public @interface ReactiveSession {
```

Comment:
I think that `@WithinReactiveSession` would be a more meaningful name... although it's quite long. OTOH it should not be necessary in most cases.

Reply:
We now have `@WithSession` (opens a new session or reuses existing) and `@WithSessionOnDemand` (opens a session or reuses existing _if needed_).